flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -321,7 +339,27 @@ private String operatorName(String suffix) {
           equalityFieldIds.add(field.fieldId());
         }
       }
-      IcebergStreamWriter<RowData> streamWriter = createStreamWriter(table, flinkRowType, equalityFieldIds);
+
+      // Fallback to use upsert mode parsed from table properties if don't specify in job level.
+      boolean upsertMode = upsert || PropertyUtil.propertyAsBoolean(table.properties(),
+          UPSERT_MODE_ENABLE, UPSERT_MODE_ENABLE_DEFAULT);
+
+      // Validate the equality fields and partition fields if we enable the upsert mode.
+      if (upsertMode) {
+        Preconditions.checkState(!overwrite,
+            "OVERWRITE mode shouldn't be enable when configuring to use UPSERT data stream.");
```

Comment:
Can you add a test that verifies the builder doesn't allow `overwrite` and `upsert` / upsertMode?

Maybe I missed it, but seems like an important thing to have a test for in case of future code refractors.

Reply:
Agreed !

Reply:
I have added a unittest to covery this, thanks for your reminder！:) @kbendick